### PR TITLE
Upgrade browserslist and webcompat data packages

### PR DESCRIPTION
**Pull request by jsnajdr**
Inspired by a recent discussions with @sgomes about polyfills, here is a PR that upgrades the relevant packages (used to determine which polyfills to use and to build them) to their latest versions.

**How to test:** I don't think anything really changes, except that we have slightly newer data about browser/feature compat.

### CODE REVIEW

**Comment by sirreal** [comment]
Thanks for working on this. It's surprising that the build breaks on this PR but wasn't broken before.

I believe the problem is due to some _module_ files being included in the _script_ build. @gziolo started to address this in https://github.com/WordPress/gutenberg/pull/65268. We may need to finish that work now 😩

**Comment by gziolo** [comment]
The issue exists for the `script-modules` config:

Screenshot 2024-10-08 at 12 05 36

> I believe the problem is due to some module files being included in the script build. @gziolo started to address this in https://github.com/WordPress/gutenberg/pull/65268. We may need to finish that work now 😩

We could verify that by limiting what gets build as scripts for the Form block:

https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/form/view.js

Then, we can refactor this block to Interactivity API and finish the cleanup started in #65268.

**Comment by gziolo** [comment]
I pushed the potential fix in https://github.com/WordPress/gutenberg/pull/65926/commits/2bc268673b2ec65641987c66b390c0904657b535 where the view script is only built for the Form block that really needs it. The rest of view scripts were obsolete as they are powered now by view script modules built with another webpack config. Let's see if it resolves the issue.

**Comment by jsnajdr** [comment]
> It's surprising that the build breaks on this PR but wasn't broken before.

Yes, I expected this to be an uneventful update, but no, we are in a rabbit hole again 😃 

There must have been some change in the compat data that specify if a given platform supports dynamic `import`s. The errors come from webpack. Webpack uses browserslist to learn about the target platform: like if it supports arrow functions etc. and then generates its runtime code accordingly.

**Comment by sirreal** [comment]
I recall very similar issues with https://github.com/WordPress/gutenberg/pull/57924 and I updated the caniuse-lite package to fix it: https://github.com/WordPress/gutenberg/pull/58087

**Comment by jsnajdr** [comment]
This is how the error arises:
- webpack tries to detect if the target environment supports the dynamic `import` feature
- it asks `browserslist` for the list of supported browsers. Our `browserslist` config says that we support browsers with >1% usage
- `browserslist` relies on `caniuse-lite` for the usage data
- in the latest version of `caniuse-lite`, the usage for the `and_uc` browser is reported as 1.14%. The previous version we were using reports only 0.91%. That means we're suddenly supporting the `and_uc` browser now!
- according to webpack the `and_uc` browser doesn't support dynamic imports

Therefore, if you claim to support the `and_uc` browser, you cannot use dynamic imports. Perfectly logical 🙂 

Fortunately, it's webpack who has wrong info about `and_uc`. We're using a version from June 2023, very old. I'll try to upgrade to the latest one, that should fix everything.

**Review by gziolo — COMMENTED** [review]
I see that updating also webpack resolved the issue. There are some snapshots to update because the hash for files calculated by webpack for files has changed.

File: tools/webpack/blocks.js [hunk 1]
```diff
@@ -54,12 +54,12 @@ const createEntrypoints = () => {
...
 	 * These paths can be matched by the regex `blockViewRegex` in order to extract
-	 * the block's filename.
+	 * the block's filename. All blocks were migrated to script modules but the Form block.
 	 *
 	 * Returns an empty array if no files were found.
 	 */
 	const blockViewScriptPaths = fastGlob.sync(
-		'./packages/block-library/build-module/**/view*.js'
+		'./packages/block-library/build-module/form/view.js'
 	);
```
Comment (sirreal) [inline tools/webpack/blocks.js:63]: @gziolo In a follow-up we can make this a `viewScriptModule` and remove this single case. It seems to be a single module with no dependencies, an easy migration.

Comment (gziolo) [reply tools/webpack/blocks.js:63]: Yes, it seems straightforward but I went with a quick change here to rule it out as a source of issue.

**Review by gziolo — APPROVED** [review]

**Comment by jsnajdr** [comment]
Thanks @sirreal for updating the snapshots!

**Comment by sgomes** [comment]
Thank you for taking care of this, @jsnajdr! 👍

**Pull request merged**